salesforce/test/metadataApi_test.go

```diff
@@ -0,0 +1,111 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestXMLData(testing *testing.T) {
+	testing.Parallel()
+
+	// Test XMLData.ToXML() with SelfClosing = false
+	xmlData := &common.XMLData{
+		XMLName:     "test",
+		Attributes:  []*common.XMLAttributes{{Key: "test", Value: "test"}},
+		Children:    []common.XMLSchema{common.XMLString("test")},
+		SelfClosing: false,
+	}
+
+	xmlStr := xmlData.ToXML()
+	if xmlStr != `<test test="test">test</test>` {
+		testing.Errorf("XMLData.ToXML() = %s; want <test test=\"test\">test</test>", xmlStr)
+	}
+
+	// Test XMLData.ToXML() with SelfClosing = false
+	xmlData = &common.XMLData{
+		XMLName:     "test",
+		Attributes:  []*common.XMLAttributes{{Key: "test", Value: "test"}},
+		Children:    []common.XMLSchema{common.XMLString("test")},
+		SelfClosing: true,
+	}
+
+	xmlStr = xmlData.ToXML()
+	if xmlStr != `<test test="test"/>` {
+		testing.Errorf("XMLData.ToXML() = %s; want <test test=\"test\"/>", xmlStr)
```

Comment:
can you use `"github.com/stretchr/testify/assert"` instead of manually doing string comparison and constructing error messages? There's a `assert.Equal` method, which we use in some tests in the server.